packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmListInternal.kt

```diff
@@ -179,6 +184,7 @@ internal fun <E : BaseRealmObject> ManagedRealmList<E>.query(
             throw IllegalArgumentException(e.message, e.cause)
         }
     }
+    if (parent == null) error("Cannot perform subqueries on non-object lists")
```

Comment:
This `parent` is only used to make a flow returned by `list.query().asFlow()` terminate when the parent is deleted. This is achieved with an `ObjectBoundQuery` binding the query to the lifetime of the `parent. 

With this update the parent of a collection could now also be another collection, but it is not clear wether such concept should follow the lifetime of the immediate parent (which could also now be another collection) or if it should follow object parent (the owning RealmObject which has the `RealmAny` "entry property"). 

While considering all this, I realised that we would never be able to express a query on a nested collections. Nested collections will always contain `RealmAny`s (generic type `T = RealmAny`) and we are not able to query these. This is already expressed in the bound of the `List<out BaseRealmObject>.query(...)` methods. 

It could be that we at some point were able to query on non-object elements, so I didn't wan't to tear out the concept of a parent, because we could reuse it in a concept of a _parent-bound-query_, so basically took this shortcut instead of trying to differentiate the collection implementations to handle both. Can try to do it, but will increase the complexity of the list implementation a bit.

Reply:
Added the above reasoning about just throwing 👍